src/v/datalake/data_writer_interface.h

```diff
@@ -36,44 +38,85 @@ struct data_writer_error_category : std::error_category {
     }
 };
 
-inline std::error_code make_error_code(data_writer_error e) noexcept {
+inline std::error_code make_error_code(writer_error e) noexcept {
     return {static_cast<int>(e), data_writer_error_category::error_category()};
 }
 
-class data_writer {
+/**
+ * Parquet writer interface. The writer should write parquet serialized data to
+ * the output stream provided during its creation.
+ */
+class parquet_writer {
+public:
+    explicit parquet_writer() = default;
+    parquet_writer(const parquet_writer&) = delete;
+    parquet_writer(parquet_writer&&) = default;
+    parquet_writer& operator=(const parquet_writer&) = delete;
+    parquet_writer& operator=(parquet_writer&&) = default;
+    virtual ~parquet_writer() = default;
+
+    virtual ss::future<writer_error>
+      add_data_struct(iceberg::struct_value, size_t) = 0;
+
+    virtual ss::future<writer_error> finish() = 0;
+};
+
+class parquet_writer_factory {
+public:
```

Comment:
I don't feel strongly about it, but WDYT about naming these something like parquet_ostream and parquet_ostream_factory? They're easy to mix up with parquet_file_writer otherwise